### PR TITLE
Add Path of Building (Community)

### PR DESCRIPTION
**Pull request by ernstp**
The upstream app is a Windows project, but it has been ported to run on Linux by other people.
This submissions is mostly based on the Arch Pkgbuild here: https://aur.archlinux.org/packages/path-of-building-community-git/

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. Link: https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/4003
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). 
The upstream project owns it.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by ernstp** [comment]
bot, build community.pathofbuilding.PathOfBuilding

File: community.pathofbuilding.PathOfBuilding.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+app-id: community.pathofbuilding.PathOfBuilding
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: pathofbuilding
+finish-args:
+  - --socket=x11
```
Comment (barthalion) [inline community.pathofbuilding.PathOfBuilding.yml:7]:
```suggestion
  - --socket=fallback-x11
```

File: community.pathofbuilding.PathOfBuilding.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        x-checker-data:
+          type: git
+          tag-pattern: "^v([\\d.]+)$"
+      - type: patch
+        path: PathOfBuilding-force-disable-devmode.patch
+
+  - name: luajit
+    buildsystem: simple
+    build-commands:
+      - make PREFIX=${FLATPAK_DEST} install -j16
```
Comment (barthalion) [inline community.pathofbuilding.PathOfBuilding.yml:37]:
```suggestion
      - make PREFIX=${FLATPAK_DEST} install -j$FLATPAK_BUILDER_N_JOBS
```

File: community.pathofbuilding.PathOfBuilding.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    sources:
+      - type: archive
+        url: http://luajit.org/download/LuaJIT-2.1.0-beta3.tar.gz
+        sha256: 1ad2e34b111c802f9d0cdf019e986909123237a28c746b21295b63c9e785d9c3
+
+  - name: Lua-cURLv3
+    buildsystem: simple
+    build-commands:
+      - make DESTDIR=${FLATPAK_DEST} LUA_IMPL=luajit -j16
```
Comment (barthalion) [inline community.pathofbuilding.PathOfBuilding.yml:46]:
```suggestion
      - make DESTDIR=${FLATPAK_DEST} LUA_IMPL=luajit -j$FLATPAK_BUILDER_N_JOBS
```

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (barthalion) [inline flathub.json:2]: I thought luajit can work with aarch64 too?

File: community.pathofbuilding.PathOfBuilding.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        url: https://github.com/Lua-cURL/Lua-cURLv3/archive/refs/tags/v0.3.13.tar.gz
+        sha256: aba40511a7cac4422c0238d1db42b2124ea5a727b0745f7f434f3dc119cbb2db
+
+  - name: pobfrontend
+    buildsystem: meson
+    post-install:
+      - install -Dm755 pobfrontend $FLATPAK_DEST/pathofbuilding
+    sources:
+      - type: git
+        url: https://gitlab.com/bcareil/pobfrontend.git
```
Comment (barthalion) [inline community.pathofbuilding.PathOfBuilding.yml:60]: Should use a commit instead of a branch.

**Comment by barthalion** [comment]
Can you also reach out to bcareil, as they're the maintainer of the fork?

**Comment by ernstp** [comment]
> Can you also reach out to bcareil, as they're the maintainer of the fork?

Right I've done that also, no reply yet: https://gitlab.com/bcareil/pobfrontend/-/issues/1

File: flathub.json (hunk 4 above)
Comment (hfiguiere) [reply flathub.json:2]: yes it does

Comment (ernstp) [reply flathub.json:2]: This is used in companion with the Windows game Path of Exile (which runs very will through Proton, hence the interest) which is x86-64 only. Also, the C++ pobfrontend implementation has never been tested on aarch64 either AFAIK, so I'm hesitant...

**Comment by ernstp** [comment]
I have addressed all the comments now @barthalion  (New season starts tomorrow so it would be fun to have this out :-) )

**Comment by barthalion** [comment]
bot, build community.pathofbuilding.PathOfBuilding

File: flathub.json (hunk 4 above)
Comment (barthalion) [reply flathub.json:2]: Please try removing this after merge.

**Pull request closed without merging**